src/System.Management.Automation/security/SecurityManager.cs

```diff
@@ -516,11 +516,23 @@ private static void UntrustPublisher(Signature signature)
             }
         }
 
+        // Check the signature via the SIP which should never erroneously validate an invalid signature
+        // or altered script.
         private static Signature GetSignatureWithEncodingRetry(string path, ExternalScriptInfo script)
         {
+            // Invoke the SIP directly with the most simple method
+            Signature signature = SignatureHelper.GetSignature(path, null);
+            if (signature.Status == SignatureStatus.Valid)
+            {
+                return signature;
+            }
+
+            // try harder to validate the signature by being explicit about encoding
+            // and providing the script contents
             string verificationContents = System.Text.Encoding.Unicode.GetString(script.OriginalEncoding.GetPreamble()) + script.ScriptContents;
```

Comment:
Can we add a `using System.Text;` to the top for this?

Reply:
done